code/renderers/vue3/template/stories/ReactiveDecorators.stories.ts

```diff
@@ -0,0 +1,161 @@
+import { userEvent, within } from '@storybook/testing-library';
+import type { Meta, StoryObj } from '@storybook/vue3';
+import { h } from 'vue';
+import { RESET_STORY_ARGS, STORY_ARGS_UPDATED, UPDATE_STORY_ARGS } from '@storybook/core-events';
+import Reactivity from './Reactivity.vue';
+
+const meta = {
+  component: Reactivity,
+  args: { label: 'If you see this, args are not updated properly' },
+  play: async ({ canvasElement, id, args }) => {
+    const channel = globalThis.__STORYBOOK_ADDONS_CHANNEL__;
+
+    const canvas = within(canvasElement);
+
+    await channel.emit(RESET_STORY_ARGS, { storyId: id });
+    await new Promise((resolve) => channel.once(STORY_ARGS_UPDATED, resolve));
+
+    const input = await canvas.findByLabelText<HTMLInputElement>('Some input:');
+    await userEvent.type(input, 'value');
+
+    await channel.emit(UPDATE_STORY_ARGS, {
+      storyId: id,
+      updatedArgs: { label: 'updated label' },
+    });
+    await new Promise((resolve) => channel.once(STORY_ARGS_UPDATED, resolve));
+
+    await expect(canvas.getByRole('button')).toHaveTextContent('updated label'); // if this passes story args are reactive
+    await expect(input).toHaveValue('value'); // if this passes story is not remounted
+  },
+} satisfies Meta<typeof Reactivity>;
+
+export default meta;
+type Story = StoryObj<typeof meta>;
+
+export const NoDecorators: Story = { args: { label: 'No decorators' } };
+
+export const DecoratorVNode: Story = {
+  decorators: [
+    (storyFn, context) => {
+      return h('div', [h('h2', ['Decorator not using args']), [h(storyFn(context.args))]]);
+    },
+  ],
+};
+
+export const DecoratorVNodeArgsFromContext: Story = {
+  decorators: [
+    (storyFn, context) => {
+      return h('div', [
+        h('h2', `Decorator use args: ${context.args.label}`),
+        [h(storyFn(context.args))],
+      ]);
+    },
+  ],
+};
+
+export const DecoratorVNodeTemplate: Story = {
+  decorators: [
+    (storyFn, context) => {
+      return h({
+        components: {
+          story: storyFn(context.args),
+        },
+        template: '<div><h2>Decorator not using args</h2><story/></div>',
+      });
+    },
+  ],
+};
+
+export const DecoratorVNodeTemplateArgsFromData: Story = {
+  decorators: [
+    (storyFn, context) => {
+      return h({
+        components: {
+          story: storyFn(context.args),
+        },
+        data() {
+          return { args: context.args };
+        },
+        template: '<div><h2>Decorator using label: {{args.label}}</h2><story/></div>',
+      });
+    },
+  ],
+};
+
+export const DecoratorVNodeTemplateArgsFromProps: Story = {
+  decorators: [
+    (storyFn, context) => {
+      return h({
+        components: {
+          story: storyFn(context.args),
+        },
+        props: ['label'],
+        template: '<div><div>Decorator using label: {{label}}</div><story/></div>',
+      });
+    },
+  ],
+};
+
+export const DecoratorFunctionalComponent: Story = {
```

Comment:
@kasperpeulen Here you want to test a functional component but you use the Input to check if state is preseved ?
Normally Functional component is pure function so there is no state or lifecycle it just props -> vnode so 100% your input loses his value 
However the component in reactive to changes and it is not remounted